lib/utils/ec2.go

```diff
@@ -109,8 +111,21 @@ func NewInstanceMetadataClient(ctx context.Context) (*InstanceMetadataClient, er
 
 // IsAvailable checks if instance metadata is available.
 func (client *InstanceMetadataClient) IsAvailable(ctx context.Context) bool {
-	_, err := client.getMetadata(ctx, "")
-	return err == nil
+	// Doing this check via imds.Client.GetMetadata() involves several unrelated requests and takes a few seconds
+	// to complete when not on EC2. This approach is faster.
+	httpClient := http.Client{
+		Timeout: 250 * time.Millisecond,
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://169.254.169.254/latest/meta-data", nil)
+	if err != nil {
+		return false
+	}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
```

Comment:
I think this should be under the actual call (ln 124) else it could return before it closes.

Reply:
`Do` only returns a body that needs closing when err is nil, so I think this should be OK.

Reply:
Ahh ok thanks, good to know.